frontend/app/src/components/SuspendedModal.svelte

```diff
@@ -0,0 +1,36 @@
+<script lang="ts">
+    import { getContext } from "svelte";
+    import ModalContent from "./ModalContent.svelte";
+    import { _ } from "svelte-i18n";
+    import type { OpenChat } from "openchat-client";
+    import Markdown from "./home/Markdown.svelte";
+
+    const client = getContext<OpenChat>("client");
+
+    function buildNoticeText(): string {
+        const suspensionDetails = client.user.suspensionDetails!;
+        const actionDate = new Date(Number(suspensionDetails.action.timestamp));
+        const actionText = suspensionDetails.action.kind === "delete_action"
+            ? "deleted"
+            : "unsuspended";
+        return `
+Your account has been suspended by @UserId(${suspensionDetails.suspendedBy} for the reason "${suspensionDetails.reason}". 
```

Comment:
Seems like an unnecessary complication telling them _who_ they have been suspended by. Particularly given that we may well end up automating suspension under some circumstances. I would just state the reason.